tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/tests/PitclipseRunConfigurationMutationTabTest.java

```diff
@@ -162,94 +144,31 @@ public void checkAllMutants() { // NOSONAR
         PAGES.getRunMenu().checkAllMutators(TEST_CONFIG_NAME);
         // run test and confirm result is as expected
         PAGES.getRunMenu().runPitWithConfiguration(TEST_CONFIG_NAME);
-        coverageReportGenerated(TESTED_CLASSES, COVERAGE, 1, 84, 1);
+        coverageReportGenerated(TESTED_CLASSES, COVERAGE, 5, 20, 1);
```

Comment:
We lose all these mutators because of the mutators removed by default in PIT?

Reply:
Looks like it, yes.  We did lose a lot of mutations operators (see [Mutations.java](https://github.com/pitest/pitclipse/pull/224/files#diff-3ca1151358cc5bcd92cc1d28f1b3b21d7449a7a8b415ae659e440c045e222867), [MutatorApiOfPitTest.java](https://github.com/pitest/pitclipse/pull/224/files#diff-04eaa9ff9dfedf8a411957728444c0354f603001a71d36ebee4a736604c3adb3) and [the list of research mutators](https://github.com/pitest/pitest-rv-plugin/tree/main/src/main/java/org/pitest/rv)). That being said I believe we should take some time to make sure I didn't miss something while upgrading.

Reply:
@echebbi So creating a new bundle for https://github.com/pitest/pitest-rv-plugin and adding it to our distribution would bring the old mutators back?

Reply:
Yes, just like we did for the `pitest-junit5` plugin. Do you think that we should integrate the `pitest-rv` plugin after all?

Reply:
@echebbi maybe we should/could. I'm a bit worried by the fact that people using Pitclipse would have more mutators than those using Maven and ignoring the new `pitest-rv`, but maybe we can rely on them knowing the Maven plugin as well.

Reply:
@echebbi yes that'd be useful; but would such information be easily (automatically) retrieved and easy to maintain?

Moreover, with `pitest-rv` we would restore the "old defaults" or not?

Reply:
> would such information be easily (automatically) retrieved and easy to maintain?

It might be possible to retrieve/infer the information through reflection. My intuition is that since we have access to the `MethodFactoryFactory` instances (each mutation operator is implemented as a child of `MethodMutatorFactory`) we should be able to get the info with something along the lines of:

```java
// maps plugin ID to the name we want to display in the table
//
Map<String, String> pluginIdToContributorName = Map.of(
    "org.pitest", "pitest",
    "org.pitest.pitest-rv-plugin", "pitest-rv"
);
Collection<MethodMutatorFactory> mutators = ServiceLoader.load(MethodMutatorFactory.class, IsolationUtils.getContextClassLoader());

for (MethodMutatorFactory mutator : mutators) {
    //
    // retrieve the contributor through reflection
    //
    Bundle bundle = FrameworkUtil.getBundle(mutator.getClass());
    //
    // update our 'Mutators' enum with the contributor name
    //
    String contributorName = pluginIdToContributorName.get(bundle.getBundleId());
    Mutators.valueOf(mutator.getName()).setContributorName(contributorName);
}
```
Or, if we want to be really extensible it should be possible to craft a solution based on a custom extension point so that external plug-ins can provide new mutators without having to change a single piece of code in Pitclipse's core. That would require some refactoring though so I believe that such a work would deserve its own PR.

> Moreover, with pitest-rv we would restore the "old defaults" or not?

We wouldn't because this group is not related to the research operators and has been completely removed. We may want to somehow support [the groups related to research operators](https://github.com/pitest/pitest-rv-plugin/blob/aa6a872fc0d75708c15e472534a441ee2d5ce185/src/main/java/org/pitest/rv/config/RVMutatorGroups.java) though.

Reply:
@echebbi thanks for the clarification, especially about the old defaults, which are now gone for good. I thought the `pitest-rv` was related to that so I was reluctant to add `pitest-rv` right away because that would have required to undo the removal of old defaults, but it's not like that :)

So, shall we address `pitest-rv` in a new PR and go on with this one?

I would also try to build with Java 21 and have a UI test with Java 21, but we can do that in another PR as well.

Reply:
> shall we address `pitest-rv` in a new PR and go on with this one?

@LorenzoBettini fine by me!